ResourceStatusSystem/Utilities/RSSCacheNoThread.py

```diff
@@ -365,15 +383,15 @@ def __getDictFromCacheMatches( cacheMatches ):
         cache dictionary of the form { ( elementName, elementType, statusType ) : status, ... }
 
 
-    :return: dict of the form { elementName : { ( elementType, statusType ) : status, ... }, ... }
+    :return: dict of the form { elementName : { statusType: status, ... }, ... }
     """
 
     result = {}
 
     for cacheKey, cacheValue in cacheMatches.iteritems():
-      elementName, elementType, statusType = cacheKey
+      elementName, _elementType, statusType = cacheKey
 
-      if not result.has_key( elementName ):
+      if elementName not in result:
         result[ elementName ] = {}
       result[ elementName ][ statusType ] = cacheValue
```

Comment:
This line may throw an exception if the previous test is `True`

Reply:
I don't understand what you mean:

```
In [21]: d = {}

In [22]: d['x'] = {}

In [23]: d
Out[23]: {'x': {}}

In [24]: d['x']['y'] = 'z'

In [25]: d
Out[25]: {'x': {'y': 'z'}}
```

Reply:
Oups, sorry... forget my comment!

Reply:
`result.setdefault( elementName, {})[statusType] = cacheValue`
is shorter

Reply:
Correct, changing.